[user] 회원 정보 수정 및 조회 구현

LimdaeIl opened the pull request:
## #️⃣연관된 이슈
- 이슈 번호: #114 

## 변경 타입
- [x] 신규 기능 추가/수정

## 변경 내용
- **to-be**(변경 후 설명을 여기에 작성)
- [x] TODO 회원 비밀번호 수정
- [x] TODO 회원 슬랙 아이디 수정
- [x] TODO 회원 고유 식별 번호로 단 건 회원 상세 정보 조회
- [x] TODO 회원 고유 식별 번호로 회원 슬랙 아이디 조회

## 체크리스트
- [x] 코드가 제대로 동작하는지 확인했습니다.
- [ ] 관련 테스트를 추가했습니다.
- [ ] 문서(코드, 주석, README 등)를 업데이트했습니다.

## 코멘트
- (추가적인 설명이나 코멘트가 필요한 경우 여기에 작성)

- save으로 올바르게 접두사 변경

Commit:
[#114][auth] refactor: save 접두사를 올바르게 수정하기

Commit:
[#114][user] 회원 비밀번호, 권한, 슬랙 수정 DTO와 서비스 구현

Commit:
[#114][user] 회원 비밀번호, 슬랙 아이디 수정 기능 구현

Commit:
[#114][user] 회원 비밀번호, 슬랙 아이디 수정 엔드포인트 구현

hp5234 commented on user/src/main/java/com/faster/user/app/user/application/usecase/UserServiceImpl.java:
```diff
@@ -10,12 +15,15 @@
 import com.faster.user.app.user.infrastructure.persistence.jpa.dto.QUserProjection;
 import com.faster.user.app.user.presentation.dto.GetAllUserResponseDto;
+import com.faster.user.app.user.presentation.dto.GetUserResponseDto;
+import com.faster.user.app.user.presentation.dto.GetUserSlackIdResponseDto;
```
Service(application 계층) 에서 presentation 계층의 DTO(GetUserResponseDto) 를 사용하는것으로 보여 추후 리펙토링이 필요할것같습니다.

LimdaeIl replied:
와 하나 또 배워갑니다..! 감사합니다! 다시 수정해서 올릴게요!

hp5234 reviewed (approved):
고생하셨습니다.

HanaHww2 commented on user/src/main/java/com/faster/user/app/user/application/usecase/UserServiceImpl.java:
```diff
@@ -36,6 +45,7 @@ public UpdateUserRoleResponseDto updateUserRoleByUserId(Long userId, UpdateUserR
     User userByUserId = getUserByUserId(userId);
 
     userByUserId.updateUserRole(requestDto.role());
+    userByUserId.createdBy(requestDto.updatedBy());
```
이 값은 수정이 안될텐데 왜 수정 처리를 해주셨을까요? 
createdBy와 updatedBy는 jpaAuditing 기능을 활용해서 헤더에 있는 유저 아이디 값을 저장하게 처리해주고 있습니다.

LimdaeIl replied:
일단 넣고 봤는데 빼도록하겠습니다! 감사합니다!

je-pa reviewed (approved):
고생하셨습니다! 수정완료되면 말씀해주세요!

HanaHww2 commented on user/src/main/java/com/faster/user/app/user/application/usecase/UserServiceImpl.java:
```diff
@@ -70,4 +80,40 @@ public DeleteUserResponseDto deleteUserByUserId(Long userId, DeleteUserRequestDt
+  @Override
+  public void updateUserPassword(Long userId, UpdateUserPasswordRequestDto requestDto) {
+    User user = getUserByUserId(userId);
+
```
유저 정보 조회하실 때, 삭제된 유저에 대한 제한이 없는 것 같습니다. jpaData 메소드에 AndDeletedAtIsNull 키워드 추가해주시거나, @SqlRestriction 키워드 참고 부탁드립니다.

HanaHww2 reviewed (approved):
수고하셨습니다. (❁´◡`❁)

Commit:
[#114][user] refactor: 각 계층마다 DTO 올바르게 수정하기

진우님이 임포트 부분으로 피드백 주신 내용을 보고, 드디어 DDD 를 아주 조금이나마 이제 코드로 어떻게 풀어야 할 지 감이 온 것 같다...!

Application 과 presentation 은 DIP 준수 했는데, Application 과 Domain 은 좀 더 풀어낼 방법을 모색해야 한다.

Commit:
[#114][user] feat: 퍼사드 패턴 구현하기

presentation 에서 PathVariable 로 받은 값을 application 에게 DIP를 준수하면서 전달해야 하는데, 컨트롤러에서 DTO에 한 개의 변수를 담고 DTO에 값을 넣는 코드보다 처음으로 퍼사드 패턴으로 사용하기를 했다.

HanaHww2 commented on user/src/main/java/com/faster/user/app/user/facade/UserFacadeImpl.java:
```diff
@@ -0,0 +1,98 @@
+import com.faster.user.app.user.application.dto.response.ADeleteUserResponseDto;
+import com.faster.user.app.user.application.usecase.UserService;
+import com.faster.user.app.user.presentation.dto.request.PDeleteUserRequestDto;
+import com.faster.user.app.user.presentation.dto.request.PUpdateUserPasswordRequestDto;
```
facade 계층에서 프레젠테이션을 참조하면 안 될 거 같은데, 혹시 어떨까요?

LimdaeIl replied:
와 .. 또 하나 배워갑니다!

Commit:
[#114][user] refactor: 컨트롤러에서 DTO 변환하도록 수정

컨트롤러에서 presentation 의 DTO를 application DTO 변환하도록 수정

Commit:
[#114][user] refactor: 애플리케이션 DTO 안에 presentation dto 직접 참조 삭제

애플리케이션 dto 안에서 프레젠테이션 dto 를 직접 참조하는 내용을 삭제한다.
